app/actions/GovernanceActions.js

```diff
@@ -102,21 +127,53 @@ const getTokenInventory = () => async (dispatch, getState) => {
   const piURL = sel.politeiaURL(getState());
   try {
     const { data } = await pi.getTokenInventory(piURL);
-    const inventoryTabs = {
-      activeVote: [],
-      abandonedVote: [],
-      finishedVote: [],
-      preVote: []
-    };
+    const inventory = updateInventoryFromApiData(data);
+
+    dispatch({ type: GETTOKEN_INVENTORY_SUCCESS, inventory });
+
+    return inventory;
+  } catch (error) {
+    dispatch({ error, type: GETTOKEN_INVENTORY_FAILED });
+    throw error;
+  }
+};
+
+export const COMPARE_INVENTORY_ATTEMPT = "COMPARE_INVENTORY_ATTEMPT";
+export const COMPARE_INVENTORY_SUCCESS = "COMPARE_INVENTORY_SUCCESS";
+export const COMPARE_INVENTORY_FAILED = "COMPARE_INVENTORY_FAILED";
+
+// compareInventory compares the state's inventory with a fresh new inventory.
+// If they are different, we get the batch of tokens and update the proposal list.
+// return the different hashes for each key if the inventories are different.
+// Return null otherwise.
+// @param inventory -> is an inventory object the same type of defaultInventory.
+export const compareInventory = () => async (dispatch, getState) => {
+  dispatch({ type: COMPARE_INVENTORY_ATTEMPT });
+  try {
+    const oldInventory = sel.inventory(getState());
+    const piURL = sel.politeiaURL(getState());
+    const { data } = await pi.getTokenInventory(piURL);
+    const inventory = updateInventoryFromApiData(data);
 
-    inventoryTabs.preVote = data.pre;
-    inventoryTabs.activeVote = data.active;
-    inventoryTabs.finishedVote = [ ...data.approved, ...data.rejected ];
-    inventoryTabs.abandonedVote = data.abandoned;
+    const inventoryDiff = getDefaultInventory();
+    let isDifferent = false;
+
+    Object.keys(oldInventory).forEach( async key => {
+      if (inventory[key].length > oldInventory[key].length) {
```

Comment:
This is broken. 

While unlikely to happen in practice, consider what would happen if (during the 60s interval between calls to this func) a new proposal entered the `preVote` list while 2 proposals left `preVote` and entered `activeVote`: the  new `preVote` list (in `inventory`) would have smaller `length` but new items (which wouldn't get picked up).

If the goal is to fetch the details of any new, as of yet unknown proposal (irrespective of which list it is on) it's probably simpler to just create (or maintain) a map of known proposal hashes (i.e. a map of proposal token => proposal) and then just check (for every token) whether that entry is filled (and if not, request it).